opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -609,7 +619,7 @@ const std::vector<int>& CpGrid::globalCell() const
         return current_view_data_ -> global_cell_;
     }
     else{
-        return this -> distributed_data_[0] ->global_cell_;
+        return this -> distributed_data_[0] -> global_cell_;
```

Comment:
Should we use the new function here, too?

Reply:
The problem is with the if-statement part, current_view_data_ is not what chooseData() "returns".
Option1: chooseData() "returns" current_view_data_ when distributed_data_.empty(), and distrubuted_data_ otherwise.
Option2: (the current one) chooseData() "returns" data_ when current_view_data_ is equal to data_.back(), distributed_data_ otherwise. 
Option3: Others ideas...

Here, we could used the new method, but checking the size of data_, or similar. Not sure if it's an improvement.

Reply:
Option 3:
chooseData stays the same and we do
```
return chooseData().back()-> global_cell_;
```
Shouldn't this work as we always want to use the leaf grid view here?

Reply:
Does this mean that distributed_data_[0] == distributed_data_.back(), when current_view_data_ != this-> data_.back().get()? If yes, then we can do your suggested modification

Reply:
Done!